lib/runtime-core/src/instance.rs

```diff
@@ -814,6 +822,89 @@ impl<'a> DynFunc<'a> {
     }
 }
 
+impl<'a> Exportable<'a> for Memory {
+    fn get_self(instance: &'a Instance, name: &str) -> Option<Self> {
+        let export_index = instance.module.info.exports.get(name)?;
+        if let ExportIndex::Memory(idx) = export_index {
+            let module = instance.module.borrow();
+            Some(instance.inner.get_memory_from_index(module, *idx))
+        } else {
+            None
+        }
+    }
+}
+
+impl<'a> Exportable<'a> for Table {
+    fn get_self(instance: &'a Instance, name: &str) -> Option<Self> {
+        let export_index = instance.module.info.exports.get(name)?;
+        if let ExportIndex::Table(idx) = export_index {
+            let module = instance.module.borrow();
+            Some(instance.inner.get_table_from_index(module, *idx))
+        } else {
+            None
+        }
+    }
+}
+
+impl<'a> Exportable<'a> for Global {
+    fn get_self(instance: &'a Instance, name: &str) -> Option<Self> {
+        let export_index = instance.module.info.exports.get(name)?;
+        if let ExportIndex::Global(idx) = export_index {
+            let module = instance.module.borrow();
+            Some(instance.inner.get_global_from_index(module, *idx))
+        } else {
+            None
+        }
+    }
+}
+
+impl<'a> Exportable<'a> for DynFunc<'a> {
+    fn get_self(instance: &'a Instance, name: &str) -> Option<Self> {
+        instance.dyn_func(name).ok()
+    }
+}
+
+impl<'a, Args: WasmTypeList, Rets: WasmTypeList> Exportable<'a> for Func<'a, Args, Rets, Wasm> {
+    fn get_self(instance: &'a Instance, name: &str) -> Option<Self> {
+        instance.func(name).ok()
+    }
+}
+
+/// `Exports` is used to get exports like [`Func`]s, [`DynFunc`]s, [`Memory`]s,
+/// [`Global`]s, and [`Table`]s from an [`Instance`].
+///
+/// Use [`Instance::exports_new`] to get an `Exports` from an [`Instance`].
+pub struct Exports<'a> {
+    instance: &'a Instance,
+}
+
+impl<'a> Exports<'a> {
+    /// Get an export.
+    ///
+    /// ```
+    /// # use wasmer_runtime_core::{DynFunc, Func, Instance};
```

Comment:
Can this be `wasmer_runtime` instead? I would prefer people depending on the the outer shell.

Reply:
Yeah, it's possible to move the doc comments with examples to `wasmer_runtime` instead, though the `#` on that line means that it's invisible to users in the docs, so it doesn't matter here